app/build.gradle

```diff
@@ -1,110 +1,149 @@
+			debuggable true
+			jniDebuggable true
+			applicationIdSuffix ".debug"
+			versionNameSuffix '-DEBUG'
+			resValue "string", "app_name", "@string/app_name_release"
+			signingConfig signingConfigs.alpha
+		}
+		alpha {
+			debuggable false
+			jniDebuggable false
+			applicationIdSuffix ".alpha"
+			versionNameSuffix '-NIGHTLY'
+			manifestPlaceholders = [ appName: "Debug" ]
+			resValue "string", "app_name", "@string/app_name_alpha"
+			signingConfig signingConfigs.alpha
+		}
+
+		dev {
+			debuggable true
+			jniDebuggable true
+			applicationIdSuffix ".dev"
+			versionNameSuffix '-dev'
+			manifestPlaceholders = [ appName: "Dev" ]
+			resValue "string", "app_name", "@string/app_name_alpha"
+			signingConfig signingConfigs.alpha
+			minifyEnabled true
+			shrinkResources true
+		}
+	}
+
+	externalNativeBuild {
+		cmake {
+			path "CMakeLists.txt"
+		}
+	}
+
+	flavorDimensions "arm"
+
+	productFlavors {
+		//        create("arm64-v8a") {
+		//            dimension("arm")
+		//            // ndk.abiFilters.add("armabi")
+		//        }
+		create( "armeabi-v7a" ) {
+			dimension( "arm" )
+			// ndk.abiFilters.add("armeabi-v7a")
+		}
+
+	}
+
+	lintOptions {
+		disable 'MissingTranslation'
+	}
+}
+
+repositories {
+	maven { url "https://jitpack.io" }
 }
 
 dependencies {
-    implementation fileTree(include: ['*.jar'], dir: 'libs')
-    androidTestImplementation('com.android.support.test.espresso:espresso-core:2.2.2', {
-        exclude group: 'com.android.support', module: 'support-annotations'
-    })
-    implementation "com.android.support:appcompat-v7:${supportLibraryVersion}"
-    implementation "com.android.support:design:${supportLibraryVersion}"
-    implementation 'com.jakewharton:butterknife:8.5.1'
-    implementation 'com.koushikdutta.ion:ion:2.+'
-    implementation "com.android.support:support-v4:${supportLibraryVersion}"
-    implementation "com.android.support:recyclerview-v7:${supportLibraryVersion}"
-    implementation "com.android.support:cardview-v7:${supportLibraryVersion}"
-    implementation 'commons-codec:commons-codec:1.5'
-    testImplementation 'junit:junit:4.12'
-    annotationProcessor 'com.jakewharton:butterknife-compiler:8.5.1'
+	implementation fileTree( include: [ '*.jar' ], dir: 'libs' )
+
+	api 'com.google.guava:guava:27.1-android'
+
+	implementation 'com.jakewharton.timber:timber:4.7.1'
+
+	implementation "androidx.core:core:1.0.2"
+	implementation "androidx.legacy:legacy-support-v4:1.0.0"
+	implementation "androidx.appcompat:appcompat:1.0.2"
+	implementation "androidx.recyclerview:recyclerview:1.0.0"
+	implementation "androidx.cardview:cardview:1.0.0"
+	implementation "androidx.constraintlayout:constraintlayout:1.1.3"
+	implementation "com.google.android.material:material:1.0.0"
+
+	implementation 'com.jakewharton:butterknife:10.1.0'
+	implementation 'com.koushikdutta.ion:ion:2.2.1'
+	implementation 'commons-codec:commons-codec:1.10'
+
+	implementation 'com.google.android.gms:play-services-base:16.1.0'
```

Comment:
We do not want to depend on any google specific services or anything preinstalled on the phone except android itself

Reply:
We want the app to be fully compatible with any AOSP phones out there

Reply:
Ok, I believe I only added that to play with AccountPicker which is uneccesary anyhow.